algorithms/src/polycommit/sonic_pc/polynomial.rs

```diff
@@ -215,9 +216,9 @@ impl<'a, F: PrimeField> LabeledPolynomialWithBasis<'a, F> {
 
     /// Compute a linear combination of the terms in `self.polynomial`, producing an iterator
     /// over polynomials of the same time.
-    pub fn sum(&self) -> impl Iterator<Item = PolynomialWithBasis<'a, F>> {
+    pub fn sum(&self) -> Result<impl Iterator<Item = PolynomialWithBasis<'a, F>>> {
         if self.polynomial.len() == 1 && self.polynomial[0].0.is_one() {
-            vec![self.polynomial[0].1.clone()].into_iter()
+            Ok(vec![self.polynomial[0].1.clone()].into_iter())
```

Comment:
Actually I wanted to do the same but because the else condition returns a Vec.into_iter(), which is a different type, we can't. Then decided to remove the entire else statement because it is dead code: https://github.com/AleoHQ/snarkVM/pull/2207

Reply:
Ah, that makes sense, I didn't notice it; since it's going to be cut out, nothing to do here.